src/freelist.rs

```diff
@@ -93,9 +94,10 @@ where
             let free = get_type_free(ty).unwrap_or_else(|| tp_free_fallback(ty));
             free(obj as *mut c_void);
 
-            #[cfg(Py_3_8)]
-            if ffi::PyType_HasFeature(ty, ffi::Py_TPFLAGS_HEAPTYPE) != 0 {
-                ffi::Py_DECREF(ty as *mut ffi::PyObject);
+            if cfg!(Py_3_8) {
```

Comment:
Why do you prefer this instead of `if cfg!(Py_3_8) && .. `?
Or, if you want to avoid `#[allow(clippy)]` you can use
```rust
            cfg_if::cfg_if! {
                if #[cfg(Py_3_8)] {
                    if ffi::PyType_HasFeature(ty, ffi::Py_TPFLAGS_HEAPTYPE) != 0 {
                        ffi::Py_DECREF(ty as *mut ffi::PyObject);
                    }
                }
            }
```
Though I'm not sure which is bettter.

Reply:
I was thinking of later, when the `#[cfg]` form can be used again, it is easier to replace this back.

In general I hate the "collapsible_if" lint, since it is in at least 50% of cases *not* an oversight and makes the code clearer.